test/address_test.rb

```diff
@@ -125,6 +125,17 @@ def initialize
 
       expect(Yast::Host.names(new_ip)).to eql original_names
     end
+
+    it "do not crash when ip was changed, no hostnames exist and the hostname is empty" do
+      new_ip = "2.2.2.2"
+
+      allow(Yast::Host)
+        .to receive(:names)
+        .and_return([])
+
+      subject.send(:update_hostname, new_ip, "")
+      expect(Yast::Host.names(new_ip)).to eql []
```

Comment:
We are expecting the same that we are allowing, so it looks like a wrong unit test :P

Reply:
hi hi forgot to save last change ... updating soon